### DISCUSSION TITLE
2023-05-29 개발일지

### DISCUSSION THREAD

**Discussion by Craftcode17** [Blue-club/pintos1#5] [개발일지]
오늘도 다들 화이팅!

**Comment by Craftcode17** [comment]
### Today's DONE : 
* 구현 완료한 Alarm Clock 코드 개선, 관련하여 팀원 토의 진행
* 개념 공부 : 우선순위 스케줄링, 동기화 기법(락, 세마포어, 조건 변수)
* Priority scheduling 요구조건 분석, 수행
    * (학습완료, 구현중) **목표1** : 스케줄링 방식 변경 (FIFO -> Priority)
    * (관련 학습 진행중) **목표2** : 동기화 premitives 구현
    * (관련 학습 진행중) **목표3** : 우선순위 donation 구현 ( priority inheritance )
* 동료학습 자료 번역 : kaist gitbook, kaist OS Lab - Thread 01 Slide ([노션 공유링크](https://quilted-class-1cc.notion.site/6058519410d542f69da46c632c65f5fa))

### Tommorow's TODO :
* priority scheduling 완료 ( ~ 18:00 )
    * **목표1** : 스케줄링 방식 변경 (FIFO -> Priority)
    * **목표2** : 동기화 premitives 구현
    * **목표3** : 우선순위 donation 구현 ( priority inheritance )
* condvar(priority-condvar) 구현
    * 조건 변수 관련학습 필요하다면 추가 수행
* Advanced scheduling 개념공부 도입
    * 개념 키워드 : Multi-Level Feedback Queue Scheduler
    * kaist gitbook 해당자료
    * OSTEP 83page, 8. 스케줄링: 멀티 레벨 피드백 큐

**Comment by Riudiu** [comment]
- alarm clock 블락 처리 및 sleep_list 활용 구현
- 동기화, 임계영역, 스핀락, 뮤텍스, 세마포어 개념 공부
